### ISSUE TITLE
regex match wrong with \s

### ISSUE THREAD

**Issue by gngshn** [ggreer/the_silver_searcher#921]
There is file named a.txt include text below(except text in brackets and brackets itself)

```
sldkf sdlfjlsdjf lsdjflsdjf sldjflsdjf
sldfkjsld lsdfl   (here have trailing whitespace)
sdlfkjsldflsdjflsdjf sldjflsdjf      (here have trailing whitespace)

sldf
```

then I execute

```
ag '\w+\s+' a.txt
ag '\w+\s+$' a.txt
```

The result is

```
$ ag '\w+\s+' a.txt 
1:sldkf sdlfjlsdjf lsdjflsdjf sldjflsdjf

$ ag '\w+\s+$' a.txt 
2:sldfkjsld lsdfl   (here have trailing whitespace)
3:sdlfkjsldflsdjflsdjf sldjflsdjf      (here have trailing whitespace)
4:
5:sldf
```

The result is wrong

**Comment by gngshn** [comment]
As my testing, it seems that ag consider '\n' as a '\s'
`ag 'ldjflsdjf\ssld'` can search line1 and 2.

**Comment by gngshn** [comment]
`ag '.*' a.txt' any output line1 in a.txt

**Comment by forgottenswitch** [comment]
This was fixed in commit `4b0b3b6b79ebf8de4dbe4583b45f92e261e3f72b`

**Comment by cup** [comment]
https://github.com/ggreer/the_silver_searcher/issues/1215
https://github.com/ggreer/the_silver_searcher/issues/459
https://github.com/ggreer/the_silver_searcher/issues/560
https://github.com/ggreer/the_silver_searcher/issues/682
https://github.com/ggreer/the_silver_searcher/issues/713

https://github.com/ggreer/the_silver_searcher/issues/921
